pkg/scheduler/preemption/preemption.go

```diff
@@ -205,6 +206,24 @@ var HumanReadablePreemptionReasons = map[string]string{
 	kueue.InCohortReclamationReason:           "reclamation within the cohort",
 	kueue.InCohortFairSharingReason:           "fair sharing within the cohort",
 	kueue.InCohortReclaimWhileBorrowingReason: "reclamation within the cohort while borrowing",
+	"": "UNKNOWN",
+}
+
+func preemptionMessage(preemptor *kueue.Workload, reason string) string {
+	var wUID, jUID string
+	if preemptor == nil || preemptor.UID == "" {
+		wUID = "UNKNOWN"
+	} else {
+		wUID = string(preemptor.UID)
+	}
+	uid, ok := preemptor.Labels[constants.JobUIDLabel]
+	if !ok || uid == "" {
+		jUID = "UNKNOWN"
+	} else {
+		jUID = uid
+	}
+
+	return fmt.Sprintf("Preempted to accommodate a workload (UID: %s, JobUID: %s) due to %s", wUID, jUID, HumanReadablePreemptionReasons[reason])
```

Comment:
The JobUID name makes me a bit hesitant as in Kueue we generally want to migrate away from Jobs, as we now support also serving workloads, such as StatefulSet, LWS, see also https://github.com/kubernetes-sigs/kueue/issues/4286 for more discussion. 

So, probably something like "OwnerUID" would be more appropriate, however, since the label is already pre-existing and named "JobUID" I think it is better to be consistent with naming. I guess we need to live in Kueue with this historical inprint. 

cc @tenzen-y